migrations/helpers/mlwh_samples_update_helper.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import logging.config
+import time
+from typing import List
+import pymongo
+import sys
+import traceback
+from crawler.constants import (
+    COLLECTION_SAMPLES,
+    FIELD_CREATED_AT,
+    MONGO_DATETIME_FORMAT,
+)
+from crawler.db import (
+    create_mongo_client,
+    get_mongo_collection,
+    get_mongo_db,
+    create_mysql_connection,
+    run_mysql_executemany_query,
+)
+from crawler.helpers import (
+    get_config,
+    map_mongo_doc_to_sql_columns,
+)
+from datetime import datetime
+from crawler.sql_queries import SQL_MLWH_MULTIPLE_INSERT
+
+def valid_datetime_string(s_datetime: str) -> bool:
+    try:
+        dt = datetime.strptime(s_datetime, MONGO_DATETIME_FORMAT)
+        if dt == None:
+            return False
+        return True
+    except Exception as e:
+        print_exception()
+        return False
+
+def print_exception() -> None:
+    print(f'An exception occurred, at {datetime.now()}')
+    e = sys.exc_info()
+    print(e[0]) # exception type
+    print(e[1]) # exception message
+    if e[2]: # traceback
+      traceback.print_tb(e[2], limit=10)
+
+def update_mlwh_with_legacy_samples(config, s_start_datetime: str = "", s_end_datetime: str = "") -> None:
+    if not valid_datetime_string(s_start_datetime):
+        print('Aborting run: Start datetime is not in the expected datetime format')
+        return
+
+    if not valid_datetime_string(s_end_datetime):
+        print('Aborting run: End datetime is not in the expected datetime format')
+        return
+
+    start_datetime = datetime.strptime(s_start_datetime, MONGO_DATETIME_FORMAT)
+    end_datetime = datetime.strptime(s_end_datetime, MONGO_DATETIME_FORMAT)
+
+    if start_datetime > end_datetime:
+        print("Aborting run: End datetime must be greater than start datetime")
+        return
+
+    print(f"Starting MLWH update process with start datetime {start_datetime} and end datetime {end_datetime}")
+
+    try:
```

Comment:
hmm, not sure about having the try / except here, what does it give us?

Reply:
Catches and prints nicely any exception thrown from the database connections.

Reply:
Hmm ok, if the exception was thrown it would print it out anyway - but maybe it's easier to read this way?